realm/realm-library/src/main/java/io/realm/RealmSchema.java

```diff
@@ -87,7 +87,7 @@ public RealmObjectSchema get(String className) {
         Set<RealmObjectSchema> schemas = new LinkedHashSet<>(tableCount);
         for (int i = 0; i < tableCount; i++) {
             String tableName = realm.sharedRealm.getTableName(i);
-            if (Table.isMetaTable(tableName)) {
+            if (!Table.isModelTable(tableName)) {
```

Comment:
Is this correct? For the old logic, the loop will be skipped if it is a primary key table.